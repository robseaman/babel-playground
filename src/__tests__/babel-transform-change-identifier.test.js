const path = require('path');
const fs = require('fs');
const babel = require('@babel/core');
const transformChangeIdentifier = require('../babel-transform-change-identifier.js');

const rootPath = (fixtureName) =>
  path.join(path.resolve(), 'src', '__tests__', '__fixtures__', fixtureName);
const inputPath = (fixtureName) => path.join(rootPath(fixtureName), 'input.js');
const optionsPath = (fixtureName) =>
  path.join(rootPath(fixtureName), 'options.js');
const readOptions = (fixtureName) => {
  let optionsString;
  try {
    optionsString = fs.readFileSync(optionsPath(fixtureName), 'utf8');
  } catch (err) {
    // no options are passed as undefined, {} would work
    if (err.code === 'ENOENT') {
      return undefined;
    }
    throw err; // any other readfile error is a concern
  }
  let options;
  return eval(`options = ${optionsString}`); // allow eval to throw it's own errors
};

const runTransform = (plugin, fixtureName) => {
  const options = readOptions(fixtureName);

  const result = babel.transformFileSync(inputPath(fixtureName), {
    plugins: [[plugin, options]],
    filename: fixtureName, // name for errors
    root: rootPath(fixtureName), // allow the fixture to contain babelrc
    configFile: false, // don't search
  });
  return result && result.code ? result.code : '';
};

describe('babel-transform-change-identifier', () => {
  ['case1', 'case2'].forEach((testCase) => {
    it(`handles ${testCase}`, () => {
      expect(
        runTransform(transformChangeIdentifier, testCase)
      ).toMatchSnapshot();
    });
  });
});
