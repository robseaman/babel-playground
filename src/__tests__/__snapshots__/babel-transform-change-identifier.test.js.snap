// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-transform-change-identifier handles case1 1`] = `
"function square(x) {
  return x * x;
}"
`;

exports[`babel-transform-change-identifier handles case2 1`] = `
"function square(num) {
  return num * num;
}"
`;
